File: bypass.cpp
// DRGcore – HieuDRG – Shadow License v99
// Compile: g++ bypass.cpp -o bypass -std=c++17 -lcurl
#include <iostream>
#include <string>
#include <regex>
#include <curl/curl.h>

static size_t WriteCB(void* contents, size_t size, size_t nmemb, std::string* userp){
    userp->append((char*)contents, size*nmemb);
    return size*nmemb;
}

std::string fetch(const std::string& url){
    CURL* curl = curl_easy_init();
    std::string read;
    if(curl){
        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCB);
        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &read);
        curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 10L);
        CURLcode res = curl_easy_perform(curl);
        curl_easy_cleanup(curl);
        if(res!=CURLE_OK) return "";
    }
    return read;
}

std::string bypassShort(const std::string& url){
    std::string body = fetch("https://api.allorigins.win/get?url="+url);
    std::regex re(R"(href=["'](https?:\/\/[^"']+)["'])");
    std::smatch m;
    if(std::regex_search(body,m,re)) return m[1];
    return "No destination found";
}

int main(int argc, char* argv[]){
    if(argc<2){
        std::cout << "Usage: ./bypass <link_or_key>\n";
        return 0;
    }
    std::string input = argv[1];
    std::string out = bypassShort(input);
    std::cout << "DRGcore result: " << out << '\n';
    return 0;
}

File: bypass.js
// DRGcore – HieuDRG – Shadow License v99
const $ = id => document.getElementById(id);

const shortDomains = [
  'link4m.com','link2m.com','yeumonney.com','nhapcode1s.com',
  'link4sub.com','linkvertise.com','linkvertise.net'
];

const robloxKeys = [
  'krnl','delta','fluxus','synapse','scriptware'
];

function detectType(input){
  const u = new URL(input.startsWith('http')? input : 'https://'+input);
  if(shortDomains.some(d => u.hostname.includes(d))) return 'short';
  if(robloxKeys.some(k => input.toLowerCase().includes(k))) return 'key';
  return 'unknown';
}

async function bypassShort(url){
  // thử fetch qua cors-anywhere fallback
  const cors = 'https://api.allorigins.win/get?url=';
  const res = await fetch(cors+encodeURIComponent(url));
  const text = await res.text();
  // regex tìm href đích
  const m = text.match(/href=["'](https?:\/\/[^"']+)["']/);
  return m ? m[1] : 'Không tìm thấy link đích';
}

function bypassKey(key){
  // fake bypass demo – thực tế cần patch binary / memory
  return `Bypassed key "${key}" -> injected=true (client-side demo only)`;
}

async function bypass(){
  const val = $('url').value.trim();
  if(!val) return $('out').textContent = 'Nhập gì đó đi!';
  const type = detectType(val);
  let result;
  if(type==='short') result = await bypassShort(val);
  else if(type==='key') result = bypassKey(val);
  else result = 'Không hỗ trợ loại này';
  $('out').textContent = result;
}
